release/config.py

```diff
@@ -0,0 +1,178 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures of toolchain configs."""
+
+import os
+
+
+class ConfigGroup(object):
+  """Group of configs that are generated by a single container.
+
+  Attributes:
+    distro: string, distro of container that generates the configs.
+    version: string, version of the configs.
+    container: string, the container registry entry of the container used to
+      generated the configs, e.g. gcr.io/cloud-marketplace/google/clang-ubuntu.
+    package: string, the Bazel package of the targets in this repo to generate
+      the configs.
+    platform_target: string, the platform target name of the corresponding RBE
+      container these configs will be used together with. This is required to
+      set value for flags:
+         --extra_execution_platforms, --host_platform and --platforms.
+      Platform: https://docs.bazel.build/versions/master/platforms.html
+    constraints: string, additional toolchain constraints needed for the cpp
+      toolchain definition.
+      Toolchain: https://docs.bazel.build/versions/master/toolchains.html
+  """
+
+  # Currently supported distros.
+  _SUPPORTED_DISTROS = ["debian8", "ubuntu16_04"]
+
+  # Map from the container distro to additional pre-defined cpp toolchain
+  # definition constraints.
+  _DISTRO_CONSTRAINTS_MAP = {
+      "debian8": ["//constraints:jessie"],
+      "ubuntu16_04": ["//constraints:xenial"]
+  }
+
+  def __init__(self, distro, version, container, package, config_types,
+               platform_target, git_root, bazel_version):
+    """Inits GroupConfig.
+
+    Args:
+      distro: string, distro of container that generates the configs.
+      version: string, version of the configs.
+      container: string, the container registry entry of the container used to
+        generated the configs, e.g.
+        gcr.io/cloud-marketplace/google/clang-ubuntu.
+      package: string, the Bazel package of the targets in this repo to generate
+        the configs.
+      config_types: types of config to generated with this container, e.g.
+        default, msan.
+      platform_target: string, the platform target name of the corresponding RBE
+        container these configs will be used together with. This is required to
+        set value for flags: --extra_execution_platforms, --host_platform and
+        --platforms. (https://docs.bazel.build/versions/master/platforms.html)
+      git_root: the absolute path of the root directory of the current
+        repository.
+      bazel_version: the version of Bazel used to generated the configs.
+
+    Returns:
+      A GroupConfig object.
```

Comment:
ConfigGroup?
Also, I'm not sure I like this name, maybe ToolchainContainerConfigs or ContainerConfigs